### PR TITLE
Add code to preserve public fields of input types to JsonSerializer

### PR DESCRIPTION
**Pull request by layomia**
Making this change now that field support has been added to `JsonSerializer` - https://github.com/dotnet/runtime/pull/36986.

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/TrimmingTests/Helper.cs [hunk 1]
```diff
@@ -90,16 +90,18 @@ public static bool AssertCollectionAndSerialize<T>(object obj, string json)
         }
     }
 
-    internal class MyClass
+    public class MyClass
```
Comment (eerhardt) [inline src/libraries/System.Text.Json/tests/TrimmingTests/Helper.cs:93]: Is this change necessary?

**Review by eerhardt — APPROVED** [review]
:shipit:

File: src/libraries/System.Text.Json/tests/TrimmingTests/Helper.cs (hunk 1 above)
Comment (layomia) [reply src/libraries/System.Text.Json/tests/TrimmingTests/Helper.cs:93]: Yes - it helps avoid `warning CS0649: Field is never assigned to, and will always have its default value`
